web/src/message_helper.ts

```diff
@@ -1,14 +1,20 @@
+import assert from "minimalistic-assert";
+
 import * as alert_words from "./alert_words";
 import * as message_store from "./message_store";
+import type {Message} from "./message_store";
 import * as message_user_ids from "./message_user_ids";
 import * as people from "./people";
 import * as pm_conversations from "./pm_conversations";
 import * as recent_senders from "./recent_senders";
 import * as stream_topic_history from "./stream_topic_history";
 import * as user_status from "./user_status";
+import type {UserStatusEmojiInfo} from "./user_status";
 import * as util from "./util";
 
-export function process_new_message(message) {
+export function process_new_message(
+    message: Message & {status_emoji_info: UserStatusEmojiInfo | undefined},
+): Message {
```

Comment:
So I'm pretty sure the correct type of this function is that it accepts a `RawMessage` and converts it into a `MessageWithBooleans`, mutating what it received in place.

To do this correctly, we likely need to make the input parameter an object called `raw_message`, and rework its core flow to set a bunch of variables and make a new object, rather than doing everything in-place. This will require a big/careful refactor.